### PR TITLE
test: improve coverage of syncer/error.go to 89.6%

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
part of #998

### Check List 

Tests 

 - Unit test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: syncer/error_test.go [hunk 1]
```diff
@@ -14,14 +14,21 @@
...
 	. "github.com/pingcap/check"
+	"github.com/pingcap/errors"
+	tmysql "github.com/pingcap/parser/mysql"
+	"github.com/pingcap/tidb/errno"
+	"github.com/pingcap/tidb/infoschema"
+	gmysql "github.com/siddontang/go-mysql/mysql"
+
 	"github.com/pingcap/dm/pkg/conn"
-	"github.com/pingcap/dm/pkg/context"
+	context2 "github.com/pingcap/dm/pkg/context"
```
Comment (csuzhangxc) [inline syncer/error_test.go:29]: why import twice?

Comment (lance6716) [reply syncer/error_test.go]: https://github.com/pingcap/dm/pull/1060/commits/f6f58ffc7c653672a119dbe114c135570aeafd66

**Review by csuzhangxc — COMMENTED** [review]
LGTM

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**